State estimation

adrianbattiston opened the pull request:
I'm not sure what this is going to do, I just want to be able to access my old commits

adrianbattiston committed:
added old state estimators

added and cleaned up both adatptive and reg EKF and SPKF

adrianbattiston committed:
updated to run

updated noise for crash and some other stuff.

adrianbattiston committed:
fixed quat conventions in SPKF's

fixed quaternion conventions to match fionna n gareths, and frame of
reference stuff. works better now. Added Comp filter, but need to fix
angular velocity frame

adrianbattiston committed:
Added Hinfinity

H-infinity works as EKF - not sure about H-inf part - so really only
added EKF ?

adrianbattiston committed:
Added full state SPKF

also added estimator IC script to vary all ICs

adrianbattiston committed:
added attitude EKF and SRSPKF (doesnt work yet)

about to redo weighting in SPKFs might need to revert

adrianbattiston committed:
Some square root versions and an optimal adaptive

Square roots run, optimal adaptive doesn't really.

fionachui commented:
did it work?

On Wed, Dec 7, 2016 at 4:36 PM, Adrian Battiston <notifications@github.com>
wrote:

> I'm not sure what this is going to do, I just want to be able to access my
> old commits
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/fionachui/Matlab-Simulator/pull/1
> Commit Summary
>
>    - added old state estimators
>    - updated to run
>    - fixed quat conventions in SPKF's
>    - Added Hinfinity
>    - Added full state SPKF
>    - added attitude EKF and SRSPKF (doesnt work yet)
>    - Some square root versions and an optimal adaptive
>
> File Changes
>
>    - *M* Display/hist2plot.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-0>
>    (125)
>    - *M* Fuzzy Logic/fuzzylogicprocess.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-1>
>    (5)
>    - *M* Impact-Identification/detectimpact.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-2>
>    (5)
>    - *A* Initialize/initAEKF.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-3>
>    (25)
>    - *A* Initialize/initAHINF.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-4>
>    (35)
>    - *A* Initialize/initASPKF.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-5>
>    (38)
>    - *A* Initialize/initASPKF_opt.asv
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-6>
>    (36)
>    - *A* Initialize/initASPKF_opt.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-7>
>    (34)
>    - *A* Initialize/initCOMP.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-8>
>    (19)
>    - *A* Initialize/initEKF.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-9>
>    (8)
>    - *A* Initialize/initEKF_att.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-10>
>    (13)
>    - *A* Initialize/initEst_sensPars.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-11>
>    (20)
>    - *A* Initialize/initHINF.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-12>
>    (18)
>    - *A* Initialize/initSE_ICs.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-13>
>    (16)
>    - *A* Initialize/initSPKF.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-14>
>    (21)
>    - *A* Initialize/initSPKF_full.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-15>
>    (24)
>    - *A* Initialize/initSRSPKF.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-16>
>    (27)
>    - *A* Initialize/initSRSPKF_full.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-17>
>    (26)
>    - *A* Initialize/initgps_baro.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-18>
>    (11)
>    - *M* Initialize/inithist.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-19>
>    (23)
>    - *M* Initialize/initsensor.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-20>
>    (35)
>    - *M* Initialize/initstate.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-21>
>    (2)
>    - *A* MathOperations/cross_mat.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-22>
>    (6)
>    - *A* MathOperations/quat2axis_angle.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-23>
>    (5)
>    - *M* MathOperations/quatmultiply.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-24>
>    (6)
>    - *A* MathOperations/vex.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-25>
>    (3)
>    - *A* Plot/RMSE_to_plot.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-26>
>    (61)
>    - *A* Plot/plot_RMSE.asv
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-27>
>    (159)
>    - *A* Plot/plot_RMSE.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-28>
>    (159)
>    - *A* Plot/plot_SE.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-29>
>    (688)
>    - *A* Plot/plot_states.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-30>
>    (146)
>    - *A* Plot/quat_error.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-31>
>    (11)
>    - *A* Plot/rmse_att.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-32>
>    (80)
>    - *A* Sensors/XYZ_to_GPS.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-33>
>    (25)
>    - *A* Sensors/initsensor_params.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-34>
>    (73)
>    - *A* Sensors/measurement_model.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-35>
>    (50)
>    - *A* Sensors/moving_avg_filt.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-36>
>    (27)
>    - *A* State_Estimation/AEKF_position.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-37>
>    (173)
>    - *A* State_Estimation/AHINF_attitude.asv
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-38>
>    (217)
>    - *A* State_Estimation/AHINF_attitude.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-39>
>    (219)
>    - *A* State_Estimation/ASPKF_attitude.asv
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-40>
>    (277)
>    - *A* State_Estimation/ASPKF_attitude.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-41>
>    (281)
>    - *A* State_Estimation/ASPKF_opt_attitude.asv
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-42>
>    (342)
>    - *A* State_Estimation/ASPKF_opt_attitude.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-43>
>    (304)
>    - *A* State_Estimation/CompFilt_attitude.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-44>
>    (81)
>    - *A* State_Estimation/EKF_attitude.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-45>
>    (178)
>    - *A* State_Estimation/EKF_position.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-46>
>    (138)
>    - *A* State_Estimation/HINF_attitude.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-47>
>    (180)
>    - *A* State_Estimation/SPKF_attitude.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-48>
>    (218)
>    - *A* State_Estimation/SPKF_full_state.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-49>
>    (237)
>    - *A* State_Estimation/SRSPKF_attitude.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-50>
>    (239)
>    - *A* State_Estimation/SRSPKF_full_state.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-51>
>    (257)
>    - *M* Update/updatehist.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-52>
>    (20)
>    - *A* run_loop.asv
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-53>
>    (59)
>    - *A* run_loop.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-54>
>    (59)
>    - *M* startsim.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-55>
>    (61)
>    - *A* startsim_trajectory.asv
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-56>
>    (340)
>    - *M* startsim_trajectory.m
>    <https://github.com/fionachui/Matlab-Simulator/pull/1/files#diff-57>
>    (203)
>
> Patch Links:
>
>    - https://github.com/fionachui/Matlab-Simulator/pull/1.patch
>    - https://github.com/fionachui/Matlab-Simulator/pull/1.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/fionachui/Matlab-Simulator/pull/1>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AJTVueSHDifr4vb53bbizAV4hb7nSLk1ks5rFybtgaJpZM4LHJAu>
> .
>



-- 
*Fiona Chui*

adrianbattiston committed:
Added norm constrained SPKF

norm constrained SPKF, got EKF position working (I think),  reverted
SPKFs back to kappa gain.

adrianbattiston committed:
ICUAS ready files

All the state estimators and their values used for ICUAS

adrianbattiston committed:
EKF to error EKF

modified EKF, and HINF to use error quaternion in the update step. This
provides a more consistent estimate. Do the same for norm-constrained
UKF?

adrianbattiston committed:
run conditions for loop (broken?)

commiting because I might need to revert one....

adrianbattiston committed:
Hamilton + Paper/Exp sims

Updated all estimators to Hamilton convention. Set up estimators to run
properly for paper sims. Tuned estimators for use with experimental data
(non magnetometer correction).

adrianbattiston committed:
push so everythings updated in order to run sim on another pc

adrianbattiston committed:
current stuff. Just staying up to date

adrianbattiston committed:
Pushing last up

Pushing last update before signing off.

adrianbattiston committed:
Just removed some old data
